cnqr/_src/convex.py

```diff
@@ -16,58 +19,82 @@
     Amos+(2017) with initialization schemes proposed by Bunne+(2022).
 
     Args:
-      dim_hidden: sequence specifying size of hidden dimensions. The
+      * dim_hidden: sequence specifying size of hidden dimensions. The
         output dimension of the last layer is 1 by default.
-      init_std: value of standard deviation of weight initialization method.
-      init_fn: choice of initialization method for weight matrices (default:
+      * dim_y: data dimensionality (default: 2).
+      * init_fn: choice of initialization method for weight matrices (default:
         `jax.nn.initializers.normal`).
-      act_fn_x: choice of activation function used in context network architecture
-      act_fn_z: choice of activation function used in convex network architecture
-      dim_data: data dimensionality (default: 2).
-      dim_context: context dimensionality (default: 1).
-      context_dense_layer: choice of layer type for the context modules
+      * epsilon_init: value of standard deviation of weight initialization method.
+      * tau_act_fn: choice of activation function used in context network architecture
+      * sigma_act_fn: choice of activation function used in convex network architecture
+      * pos_act_fn: choice of positive activation function (default: `nn.relu`)
+      * context_dense_layer: choice of layer type for the context modules
         (default: nn.Dense, later BjorckDense might be used)
-      gaussian_map: data inputs of source and target measures for
+      * gaussian_map: data inputs of source and target measures for
         initialization scheme based on Gaussian approximation of input and
         target measure (if None, identity initialization is used).
     """
 
     dim_hidden: Sequence[int]
-    init_std: float = 1e-1
+    dim_y: int = 2
     init_fn: Callable = jax.nn.initializers.normal
-    act_fn_x: Callable = nn.softplus  # alternative -> relu
-    act_fn_z: Callable = nn.softplus  # alternative -> relu
-    dim_data: int = 2
-    dim_context: int = 1
-    context_dense_layer: str = nn.Dense  # TODO: try with BjorckDense
+    epsilon_init: float = 1e-2
+    tau_act_fn: Callable = nn.softplus  # alternative -> GroupSort
+    sigma_act_fn: Callable = nn.softplus  # alternative -> ReLU, ELU
+    pos_act_fn: Callable = nn.relu
+    context_dense_layer: nn.Module = nn.Dense  # TODO: try with BjorckDense
     gaussian_map: Tuple[jnp.ndarray, jnp.ndarray] = None
 
-    @staticmethod
-    def _compute_identity_map(input_dim):
-        A = jnp.eye(input_dim).reshape((1, input_dim, input_dim))
-        b = jnp.zeros((1, input_dim))
-        return A, b
-
-    @staticmethod
-    def _identity_initializer(key: KeyArray,
-                              shape: core.Shape,
-                              dtype: DTypeLikeInexact = jnp.float_) -> Array:
-        return jnp.eye(N=shape[0], M=shape[1], dtype=dtype)
-
     @nn.compact
     def __call__(self, x, y):
         """
```

Comment:
Please document the expected shape of input and of return of the `__call__` method like I did in ` BjorckDense`. Specify the batch in every case for `x` and `y`.